### PR TITLE
[5.0] Use hazelcast.jar's NOTICE file as the the distribution NOTICE

### PR DESCRIPTION
**Pull request by kwart**
This PR replaces the outdated static `NOTICE` file in the ZIP distribution with the generated `THIRD-PARTY.txt` and the `NOTICE` file from `hazelcast.jar`

### CODE REVIEW

File: distribution/pom.xml [hunk 1]
```diff
@@ -67,7 +67,7 @@
                                     <groupId>com.hazelcast</groupId>
                                     <artifactId>hazelcast</artifactId>
                                     <outputDirectory>${configs.from.jar}</outputDirectory>
-                                    <includes>**/*.xml,**/*.yaml</includes>
+                                    <includes>**/*.xml,**/*.yaml,META-INF/NOTICE</includes>
```
Comment (kwart) [inline distribution/pom.xml:70]: Another option here could be including the `NOTICE` file from the GH repo root. I don't know which is a better option.

**Review by olukas — APPROVED** [review]

File: distribution/pom.xml (hunk 1 above)
Comment (frant-hartm) [reply distribution/pom.xml:70]: The difference is that this one (the one from hazelcast.jar) contains our NOTICE file (from GH repo root) appended with NOTICE files of shaded dependencies of the `hazelcast` module - jackson in this case.

I think including that info is more correct. I would expect this should include also NOTICE files from shaded dependencies in the extension jars. Let's discuss this with jiri and improve in next release.

File: distribution/src/assembly/assembly-descriptor-slim.xml [hunk 2]
```diff
@@ -48,7 +48,7 @@
         <file>
             <source>${project.build.directory}/classes/META-INF/THIRD-PARTY.txt</source>
             <destName>THIRD-PARTY.txt</destName>
-            <outputDirectory>licenses</outputDirectory>
+            <outputDirectory>/</outputDirectory>
```
Comment (frant-hartm) [inline distribution/src/assembly/assembly-descriptor-slim.xml:51]: Is there a requirement for the THIRD-PARTY.txt to be in root? It's polluting the root folder with yet another useless file.

File: distribution/src/assembly/assembly-descriptor-slim.xml [hunk 3]
```diff
@@ -128,6 +128,13 @@
             <destName>hazelcast-security-hardened.xml</destName>
             <outputDirectory>config/examples</outputDirectory>
         </file>
+
+        <!-- NOTICE file from hazelcast.jar -->
+        <file>
+            <source>${configs.from.jar}/META-INF/NOTICE</source>
```
Comment (frant-hartm) [inline distribution/src/assembly/assembly-descriptor-slim.xml:134]: We could rename this property to `files.from.jar` but that's really nit picking :-)

**Review by frant-hartm — APPROVED** [review]
All minor comments, but if that `THIRD-PARTY.txt` really doesn't need to be moved to root I wouldn't move it.

File: distribution/src/assembly/assembly-descriptor-slim.xml (hunk 2 above)
Comment (kwart) [reply distribution/src/assembly/assembly-descriptor-slim.xml]: reverted

**Pull request merged**